### PR TITLE
Ds expand lua deepbinding

### PR DESCRIPTION
**Pull request by XT3RM1N8R**
playerData:
color
daysLeftWithoutCastle

town:
x
y
visitingHeroIdx
visitingHero

hero:
tempMoraleBonuses
tempLuckBonuses
mobility
remainingMobility
x
y

army:
creatureIdx
hex
initialQuantity
quantity
morale
luck

### CODE REVIEW

File: maps/MODULES/binding.lua [hunk 1]
```diff
@@ -46,18 +66,39 @@ town_mt = {
 			return GetTownOwner(t)
 		elseif k == "faction" then
 			return GetTownFaction(t)
+		elseif k == "x" then
+			return GetTownX(t)
+		elseif k == "y" then
+			return GetTownY(t)
+		elseif k == "visitingHeroIdx" then
```
Comment (jkoppel) [inline maps/MODULES/binding.lua]: I'm guessing this is one of the ones you did while sleep deprived. One thing that happens on all-nighters is it's easy to just sit down and do things without without asking why or whether you need to.

Remove this one. As I've said, indices like this should not be exposed to the scripter.

File: maps/MODULES/binding.lua [hunk 2]
```diff
@@ -80,10 +127,18 @@ battleStack_mt = {
 			return GetStackSide(t)
 		elseif k == "type" then
 			return GetStackType(t)
+		elseif k == "creatureIdx" then
```
Comment (jkoppel) [inline maps/MODULES/binding.lua]: Rename to something like "creatureType" .

File: maps/MODULES/binding.lua (hunk 1 above)
Comment (XT3RM1N8R) [inline maps/MODULES/binding.lua]: Oh, I think I forgot that you said this. My bad. Had you not reminded me here, I don't think I would have remembered. xD

File: maps/MODULES/binding.lua (hunk 2 above)
Comment (XT3RM1N8R) [inline maps/MODULES/binding.lua]: I used creatureIdx because this was the actual name of the property, but I will rename it as "creatureType"

**Pull request merged**